src/laddie/laddie_main.f90

```diff
@@ -0,0 +1,82 @@
+MODULE laddie_main
+
+  ! The main laddie model module.
+
+! ===== Preamble =====
+! ====================
+    
+  USE precisions                                             , ONLY: dp
+  USE mpi_basic                                              , ONLY: par, sync
+  USE control_resources_and_error_messaging                  , ONLY: crash, init_routine, finalise_routine, colour_string
+  USE model_configuration                                    , ONLY: C
+  USE parameters
+  USE mesh_types                                             , ONLY: type_mesh
+  USE ice_model_types                                        , ONLY: type_ice_model
+  USE laddie_model_types                                     , ONLY: type_laddie_model
+  USE ocean_model_types                                      , ONLY: type_ocean_model
+  USE reallocate_mod                                         , ONLY: reallocate_bounds
+  USE laddie_utilities                                       , ONLY: compute_ambient_TS, allocate_laddie_model
+
+  IMPLICIT NONE
+    
+CONTAINS
+    
+! ===== Main routines =====
+! =========================
+  SUBROUTINE initialise_laddie_model( mesh, laddie, ocean, ice)
+    ! Initialise the laddie model
+
+    ! In- and output variables
+
+    TYPE(type_mesh),                        INTENT(IN)    :: mesh
+    TYPE(type_laddie_model),                INTENT(INOUT) :: laddie
+    TYPE(type_ocean_model),                 INTENT(IN)    :: ocean
+    TYPE(type_ice_model),                   INTENT(IN)    :: ice
+
+    ! Local variables:
+    CHARACTER(LEN=256), PARAMETER                         :: routine_name = 'initialise_laddie_model'
+    INTEGER                                               :: vi
+    REAL(dp), PARAMETER                                   :: H_init = 10.0_dp ! [m]    Initial thickness
+    REAL(dp), PARAMETER                                   :: T_off  = 0.0_dp  ! [degC] Initial temperature offset
+    REAL(dp), PARAMETER                                   :: S_off  = -0.1_dp ! [PSU]  Initial salinity offset
```

Comment:
Should these three parameters be configurable?

Reply:
They're not very relevant for the model outcome and these values generally lead to a smooth spinup. So to not over-clutter the config file, I put them here.

If you prefer them as config parameters, I can add them there. Up to you

Reply:
If you expect anyone to ever want to change them, then I'd prefer them as config variables. If not, then leave them here.

Reply:
OK I'll move them to config